### DISCUSSION TITLE
Support for time based habits

### DISCUSSION THREAD

**Discussion by Nextross** [iSoron/uhabits#1110] [Feature Requests]
Something like from "I am sober". Just a timer and maybe a statistic when you break the habit.

**Comment by hiqua** [comment]
It's unclear to me what you mean, can you expand on your example, on why the current version does not match your need?

**Reply by Nextross** [reply]
I'll try.

So the current version have 2 options how to track your habits:

Yes or No question (habits) - for example "Have you played piano today?"
Measureable habits - for example "How many miles did you run today?"

What I'm missing (or can't find anywhere) is a third option where you could set a date (let's say 20/09/2021) and a timer (clock) would track how many seconds, minutes, hours, days, months, years have passed since that date. For example you would name your habit "Self-harm free" and after click the habit, it would display a timer with a (not important) text, something like "I've been self-harm free for xyz days/months etc."

**Reply by hiqua** [reply]
Ok, but I think this is already somewhat covered by the current implementation: if you have "Have you been sober today" as a habit, then clicking it from the main screen will show you the streaks and how long the current streak has been.

Why is this not a good solution for your examples?

Maybe related: https://github.com/iSoron/uhabits/discussions/696

**Reply by Nextross** [reply]
That works. I might have overlooked this. But I think it's more convenient to set a habit and "forget" about it and when you want to, you can just see the exact time

**Reply by hiqua** [reply]
As in, having a default value ("being sober") and just opening the app when this default value is not right ("not being sober"), is this what you mean?

**Reply by Nextross** [reply]
Probably not. Can you please elaborate?

**Reply by hiqua** [reply]
Would that be enough if you could make it so that a habit is checked by default?

**Reply by Nextross** [reply]
Probably yes. It's just more enjoyable to see the timer going, I guess

**Reply by iSoron** [reply]
See also #125.
